### PR TITLE
feat(HDWallet): add HD Wallet Support

### PR DESCRIPTION
**Pull request by shekhar-shubhendu**

### CODE REVIEW

**Comment by shekhar-shubhendu** [comment]
closes #280

**Comment by shekhar-shubhendu** [comment]
add test cases from https://github.com/satoshilabs/slips/blob/master/slip-0010.md

File: aeternity/hdwallet.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    @staticmethod
+    def from_keystore(path, password=""):
+        with open(path) as fp:
+            keystore = json.load(fp)
+        if keystore.get("crypto", {}).get("secret_type") == SECRET_TYPE_BIP39:
+            mnemonic = keystore_open(keystore, password).decode()
+            return HDWallet(mnemonic)
+        else:
+            raise TypeError("Invalid keystore")
```
Comment (noandrea) [inline aeternity/hdwallet.py]: probably better to change the error message to
The keystore is not for a mnemonic secret

Comment (shekhar-shubhendu) [reply aeternity/hdwallet.py]: Then there should be two:
- invalid keystore: when file itself is incompatible.
- wrong keystore type: when user provides keystore of masterkey.

**Pull request merged**